src/lib/components/Swap/RoutingDiagram/index.tsx

```diff
@@ -0,0 +1,160 @@
+import { Currency, TradeType } from '@uniswap/sdk-core'
+import { FeeAmount } from '@uniswap/v3-sdk'
+import { ReactComponent as AutoRouterIcon } from 'assets/svg/auto_router.svg'
+import { ReactComponent as DotLine } from 'assets/svg/dot_line.svg'
+import Column from 'lib/components/Column'
+import Row from 'lib/components/Row'
+import Rule from 'lib/components/Rule'
+import TokenImg from 'lib/components/TokenImg'
+import styled, { ThemedText } from 'lib/theme'
+import { useMemo } from 'react'
+import { InterfaceTrade } from 'state/routing/types'
+
+import { getTokenPath, RoutingDiagramEntry } from './utils'
+
+const Wrapper = styled(Column)`
+  padding: 0.25em;
+`
+
+const RouteRow = styled(Row)`
+  grid-template-columns: 1em 1.15em 1fr 1em;
+  min-width: 430px;
+`
+
+const RouteDetailsContainer = styled(Row)`
+  padding: 0.1rem 0.5rem;
+  position: relative;
+`
+
+const ShortDottedLine = styled.div`
+  align-items: center;
+  display: flex;
+  opacity: 0.5;
+  overflow: hidden;
+  width: 71px;
+  z-index: 1;
+`
+
+const DottedLine = styled.div`
+  align-items: center;
+  display: flex;
+  opacity: 0.5;
+  position: absolute;
+  width: calc(100% - 1em);
+  z-index: 1;
+`
+
+const DotColor = styled(DotLine)`
+  path {
+    stroke: ${({ theme }) => theme.secondary};
+  }
+`
+
+const BaseBadge = styled(Row)`
+  background-color: ${({ theme }) => theme.outline};
+  border-radius: 0.5em;
+  grid-gap: 0.375em;
+  padding: 0.25em 0.375em;
+  z-index: 2; // To cover the dotted line.
+`
+
+// Used to cutoff space between badges and dotted lines.
+const TransparentBadgeWrapper = styled(BaseBadge)`
+  background-color: ${({ theme }) => theme.dialog};
+  padding: 0 4px;
+`
+
+const VersionBadge = styled(BaseBadge)`
+  background-color: ${({ theme }) => theme.module};
+  border-radius: 0.25em;
+  padding: 0 0.125em;
+`
+
+const DetailsRow = styled(Row)`
+  display: grid;
+  grid-template-columns: 4.8125em 1fr;
+  width: 100%;
+`
+
+const StyledAutoRouterLabel = styled(ThemedText.Body1)`
+  color: #27ae60;
```

Comment:
can this be a theme color?

Reply:
do we want to start defining colors like we do in app? 

for example: green1, blue1, red1 etc

Reply:
we don't yet have a large palette for widget. If you can't use the gradient, could this default to an existing color (like "primary")? Seems strange to have a single green header. The gradient makes more sense to me because it mirrors the colors used by the chips. Might be a phil question.

Reply:
Updated to use primary as a backup